pkgs/development/interpreters/ruby/default.nix

```diff
@@ -187,6 +187,11 @@ let
               sed -i '/CC_VERSION_MESSAGE/d' $rbConfig
             ''
           }
+
+          # Allow to override compiler. This is important for cross compiling as
+          # we need to set a compiler that is different from the build one.
+          sed -i 's/CONFIG\["CC"\] = "\(.*\)"/CONFIG["CC"] = ENV["CC"] || "\1"/'  "$rbConfig"
```

Comment:
We cannot make this a patch to avoid it silently breaking?

Reply:
No because the value for CC is dynamic.